### DISCUSSION TITLE
Implement a message channel

### DISCUSSION THREAD

**Discussion by nextfool** [microsoft/ClearScriptWorkerSample#5] [Ideas]
Hi @ClearScriptLib ,
I am trying to implement the MessageChannel[https://developer.mozilla.org/en-US/docs/Web/API/MessageChannel](url). I am thinking of using the new System.Threading.Channels api, but it seems a bit overkill. I also want to impalement the "transferable" behavior of the MessaagePort. Can you give me some advice on this toptic?

Thank you.

**Comment by ClearScriptLib** [comment]
Hi @nextfool,

Sorry about the delayed response.

It looks like `MessageChannel` provides a mechanism similar to Web Worker communications (`postMessage` + event callbacks), so this sample project should give you some idea. You can use `SharedArrayBuffer` as a basis for transferable objects in ClearScript 7.2 or later.

However, it may be easier to implement a shared facility as a .NET object or type, which you can drop into multiple V8 runtimes without worrying about transferability.

Good luck!
